plugins/zify/zify.ml

```diff
@@ -0,0 +1,1158 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *   INRIA, CNRS and contributors - Copyright 1999-2019       *)
+(* <O___,, *       (see CREDITS file for the list of authors)           *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+open Constr
+open Names
+open Pp
+open Lazy
+
+(* From Coqlib *)
+let make_dir l = DirPath.make (List.rev_map Id.of_string l)
+
+(** [get_type_of] performs beta reduction ;
+    Is it ok for Retyping.get_type_of (Zpower_nat n q) to return (fun _ : nat => Z) q ? *)
+let get_type_of env evd e =
+  Tacred.cbv_beta env evd (Retyping.get_type_of env evd e)
+
+(** [unsafe_to_constr c] returns a [Constr.t] without considering an evar_map.
+    This is useful for calling Constr.hash *)
+let unsafe_to_constr c = EConstr.to_constr ~abort_on_undefined_evars:false Evd.empty c
```

Comment:
Why not use
```suggestion
let unsafe_to_constr c = EConstr.Unsafe.to_constr
```
?
The way you have it seems just pointlessly slower (it will go through the whole term, but not substitute anything since it's Evd.empty and not raise errors since it's abort:false)

Reply:
Thanks for the advice.